src/scripts/slack_id_lookup_table_scripts/lookup_table_functions.ts

```diff
@@ -1,107 +1,73 @@
-import { readFile } from 'fs'
-import { exec } from 'child_process'
+import { readFileSync } from 'fs'
+import { execSync } from 'child_process'
 import { SlackUser, SlackGroup, SlackUsersResponseObject, SlackGroupsResponseObject } from './interfaces'
 
-export async function PopulateLookupTable(_lookupTable:string[], _writeLookupTableShellScriptFilePath:string, _lookupTableFilePath:string, _getSlackUserShellScriptFilePath:string, _slackUserInfoFilePath:string, _slackGroupInfoFilePath:string) {
-    await AddUserDataToLookupTable(_lookupTable, _writeLookupTableShellScriptFilePath, _lookupTableFilePath, _slackUserInfoFilePath);
-    await AddUserGroupDataToLookupTable(_lookupTable, _writeLookupTableShellScriptFilePath, _lookupTableFilePath, _slackGroupInfoFilePath);
+export function PopulateLookupTable(_lookupTable:string[], _writeLookupTableShellScriptFilePath:string, _lookupTableFilePath:string, _getSlackUserShellScriptFilePath:string, _slackUserInfoFilePath:string, _slackGroupInfoFilePath:string):void {
+    AddUserDataToLookupTable(_lookupTable, _slackUserInfoFilePath);
+    AddUserGroupDataToLookupTable(_lookupTable, _slackGroupInfoFilePath);
+    WriteLookupTableToFile(_lookupTable, _writeLookupTableShellScriptFilePath, _lookupTableFilePath);
```

Comment:
A try/catch block is recommended here in case there are any errors reading the file, parsing the JSON, or executing the bash script to write the file. Without catching the exception, the process will exit out with an `UncaughtErrorException` detailing the uncaught error.

Alternatively a try/catch block can be put around each call to `readFileSync`, `JSON.parse`, and `execSync` calls to get finer control of error output.

```suggestion
    try {
        AddUserDataToLookupTable(_lookupTable, _slackUserInfoFilePath);
        AddUserGroupDataToLookupTable(_lookupTable, _slackGroupInfoFilePath);
        WriteLookupTableToFile(_lookupTable, _writeLookupTableShellScriptFilePath, _lookupTableFilePath);
    } catch (e) {
        console.error('Something went wrong!', e);
    }
```

Reply:
Great idea, I'll add a try-catch block to each critical point with a detailed error message.